### PR TITLE
add empty line

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,10 +3,82 @@
...
+            <version>5.4.5.Final</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.5.Final</version>
+        </dependency>
 
+        <dependency>
```
Comment (dariiabutorina) [inline pom.xml]:
```suggestion
        </dependency>
        <dependency>
```

File: pom.xml [hunk 2]
```diff
@@ -3,10 +3,82 @@
...
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.21</version>
+        </dependency>
+    </dependencies>
 
-</project>
\ No newline at end of file
+    <build>
```
Comment (dariiabutorina) [inline pom.xml:35]: Let's remove the empty lines here

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public class Main {
+    private static Injector injector = Injector.getInstance("com.dev.cinema");
+
+    public static void main(String[] args) {
+        Movie movie = new Movie();
+        movie.setTitle("Fast and Furious");
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
+        movieService.add(movie);
+
+        movieService.getAll().forEach(System.out::println);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/Main.java]:
```suggestion
        movieService.add(movie);
        movieService.getAll().forEach(System.out::println);
```

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java:15]: Is transaction autoclosable?

File: src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="connection.username">root</property>
+        <property name="connection.password">1234</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="com.dev.cinema.model.Movie"/>
+    </session-factory>
+
+</hibernate-configuration>
```
Comment (dariiabutorina) [inline src/main/resources/hibernate.cfg.xml]: Empty line

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie entity", e);
+        }
+    }
+
+    @Override
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new DataProcessingException("Error creating Session Factory", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java]: its a different kind of exception, let's leave just RuntimeException here

File: pom.xml (hunk 1 above)
Comment (frostpv) [reply pom.xml]: fixed

File: pom.xml (hunk 2 above)
Comment (frostpv) [reply pom.xml:35]: fixed

File: src/main/java/com/dev/cinema/Main.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/Main.java]: fixed

File: src/main/resources/hibernate.cfg.xml (hunk 5 above)
Comment (frostpv) [reply src/main/resources/hibernate.cfg.xml]: fixed

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java (hunk 6 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/util/HibernateUtil.java (hunk 8 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/util/HibernateUtil.java]: fixed

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java (hunk 7 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: fixed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**